ssh/tailssh/testcontainers/Dockerfile

```
@@ -1,19 +1,22 @@
 ARG BASE
 FROM ${BASE}
 
+ARG BASE
+
 RUN echo "Install openssh, needed for scp."
-RUN apt-get update -y && apt-get install -y openssh-client
+RUN if echo "$BASE" | grep "ubuntu:"; then apt-get update -y && apt-get install -y openssh-client; fi
+RUN if echo "$BASE" | grep "alpine:"; then apk add openssh; fi
 
-RUN groupadd -g 10000 groupone
-RUN groupadd -g 10001 grouptwo
-# Note - we do not create the user's home directory, pam_mkhomedir will do that
+# Note - on Ubuntu, we do not create the user's home directory, pam_mkhomedir will do that
 # for us, and we want to test that PAM gets triggered by Tailscale SSH.
-RUN useradd -g 10000 -G 10001 -u 10002 testuser
+RUN if echo "$BASE" | grep "ubuntu:"; then groupadd -g 10000 groupone && groupadd -g 10001 grouptwo && useradd -g 10000 -G 10001 -u 10002 testuser; fi
+# On Alpine, we can't configure pam_mkhomdir, so go ahead and create home directory.
+RUN if echo "$BASE" | grep "alpine:"; then addgroup -g 10000 groupone && addgroup -g 10001 grouptwo && adduser -u 10002 -D testuser && addgroup testuser groupone && addgroup testuser grouptwo; fi
 
 RUN echo "Set up pam_mkhomedir."
-RUN sed -i -e 's/Default: no/Default: yes/g' /usr/share/pam-configs/mkhomedir || echo "might not be ubuntu"
-RUN cat /usr/share/pam-configs/mkhomedir
-RUN pam-auth-update --enable mkhomedir
+RUN if echo "$BASE" | grep "ubuntu:"; then sed -i -e 's/Default: no/Default: yes/g' /usr/share/pam-configs/mkhomedir; fi
+RUN if echo "$BASE" | grep "ubuntu:"; then cat /usr/share/pam-configs/mkhomedir; fi
+RUN if echo "$BASE" | grep "ubuntu:"; then pam-auth-update --enable mkhomedir; fi
```

this would be somewhat cleaner with multiline RUN commands like this:
```suggestion
RUN if echo "$BASE" | grep "ubuntu:"; then \
  sed -i -e 's/Default: no/Default: yes/g' /usr/share/pam-configs/mkhomedir && \
  cat /usr/share/pam-configs/mkhomedir && \
  pam-auth-update --enable mkhomedir; \
  fi
```

---

Good idea on the multi-line statements. I applied this in a few places.